### PR TITLE
(DOCSP-35188): C++: Close a realm

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-35188

- [Configure & Open a Realm](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-35188/sdk/cpp/realm-files/configure-and-open-a-realm/#close-a-realm): New "Close a Realm" section with code example
- [Quick Start](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-35188/sdk/cpp/quick-start/#close-a-realm): New "Close a Realm" section with code example

### Reminder Checklist

Before merging your PR, make sure to check a few things.

- [ ] Did you tag pages appropriately?
  - genre
  - meta.keywords
  - meta.description
- [x] Describe your PR's changes in the Release Notes section
- [x] Create a Jira ticket for related docs-app-services work, if any

### Release Notes

- **C++ SDK**
  - Quick Start: New "Close a Realm" section with Bluehawked, tested code example.
  - Realm Files/Configure & Open a Realm: New "Close a Realm" section with Bluehawked, tested code example.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/cpp/realm-files/configure-and-open-a-realm.txt [hunk 1]
```diff
@@ -313,3 +313,14 @@ Provide a Subset of Models to a Realm
...
       .. literalinclude:: /examples/generated/cpp/open-realm.snippet.open-default-realm.cpp
          :language: cpp
+
+.. _cpp-close-realm:
+
+Close a Realm
+-------------
+
+To avoid memory leaks, close the database when you're done using it. Closing
+the database invalidates realm objects. Close the database with ``db::close()``.
```
Comment (cbullinger) [inline source/sdk/cpp/realm-files/configure-and-open-a-realm.txt:323]: are we calling them "lowercase r" `realm objects` in this post-rename world?

**Review by cbullinger — APPROVED** [review]
lgtm!

File: source/sdk/cpp/realm-files/configure-and-open-a-realm.txt (hunk 1 above)
Comment (dacharyc) [reply source/sdk/cpp/realm-files/configure-and-open-a-realm.txt]: Good question! I didn't quite know how to handle this so I reworded it on the quick start page. Just removed this reference. I guess if we _have_ to refer to objects managed by the database, we should use the proper name - i.e. Realm. Or say "objects managed by the database?"

I don't think I have brain to do renaming stuff _and_ the GA stuff at the same time ;P

Comment (cbullinger) [reply source/sdk/cpp/realm-files/configure-and-open-a-realm.txt]: I don't have the brain for this either, so I'll add it the list of things for us to hash out before we dive into the Great Renaming

**Pull request merged**